### PR TITLE
[Annotation plugin] Rename annotation classes

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>[Annotation plugin] Rename annotation classes</changelog>`.

### Summary of changes
This pr rename annotation classes with the following rule:

- Symbol -> PointAnnotation
- Circle -> CircleAnnotation
- Line -> PolylineAnnotation
- Fill -> PolygonAnnotation

### CODE REVIEW

File: app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt [hunk 1]
```diff
@@ -34,8 +31,8 @@ class UpdateAnnotationTest : BaseMapTest(), OnMapLoadedListener {
   private lateinit var runnable: Runnable
   private var index = 0
   private val latch = CountDownLatch(AnnotationUtils.STYLES.size * 3)
-  private lateinit var symbolManager: SymbolManager
-  private lateinit var symbol: Symbol
+  private lateinit var symbolManager: PointAnnotationManager
```
Comment (tobrun) [inline app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:34]: shall we update symbolManager to be pointManager?

**Review by tobrun — APPROVED** [review]
amazing refactor!

**Review by pengdev — APPROVED** [review]

**Comment by pengdev** [comment]
Great work @Chaoba 👍 , please also update the downstream benchmark.

File: app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt (hunk 1 above)
Comment (Chaoba) [reply app/src/androidTest/java/com/mapbox/maps/testapp/annotation/UpdateAnnotationTest.kt:34]: Correct.

**Pull request merged**